vulkan_generator/generator.py

```diff
@@ -57,8 +61,32 @@ def print_vulkan_metaadata(vulkan_metadata: types.AllVulkanTypes) -> None:
     pretty_printer.pprint(vulkan_metadata.funcpointers)
 
 
-def generate(vulkan_xml_path: Path) -> bool:
+def basic_generate(target: str,
+                   output_dir: Path,
+                   all_vulkan_types: types.AllVulkanTypes,
+                   generate_header,
+                   generate_cpp,
+                   generate_test):
+
+    generate_header(os.path.join(output_dir, target + ".h"), all_vulkan_types)
+    generate_cpp(os.path.join(output_dir, target + ".cc"), all_vulkan_types)
+    generate_test(os.path.join(output_dir, target + "_tests.cpp"), all_vulkan_types)
+
+def generate(target: str, output_dir: Path, vulkan_xml_path: Path) -> bool:
+
     """ Generator function """
     all_vulkan_types = vulkan_parser.parse(vulkan_xml_path)
     print_vulkan_metaadata(all_vulkan_types)
+
+    os.makedirs(output_dir, exist_ok=True)
+
+    """ Switch table for generate target. Add new targets here and throw exception for unknown targets """
```

Comment:
This is a doc string not comment. Comments are "#"

Reply:
Done